File: backend/api_server/api/src/main/java/com/example/news/api/dto/internal/DetailedNews.java
package com.example.news.api.dto.internal;

import lombok.*;

import java.math.BigDecimal;
import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class DetailedNews {
    private int id;
    private String title;
    private Timestamp publishDate;
    private String link;
    private String language;
    private String fullText;
    private String summary;
    private String sentimentLabel;
    private BigDecimal sentiment;
    private Integer topicId;
    private Integer sourceId;
    private String topic_name;
    private String source_name;
    private List<DetailedEntity> entities = new ArrayList<>();

}

File: frontend/ai-news-ui/src/shared/types/news/DetailedNews.ts
interface EntityType {
  id: number;
  name: string;
}

interface Entity{
  id: number;
  value:string;
  type:EntityType
}

interface Keyphrase {
  id: number;
  value:string;
}

export interface DetailedNews {
  id: string;
  title: string;
  publishDate: string;
  link: string;
  language: string;
  fullText: string;
  source: {
    id: number;
    name: string;
  };
  inference: {
    summary: string;
    sentiment: {
      label: string;
      score: number;
    };
    topic: {
      id: number;
      name: string;
    };
    keyphrases: Keyphrase[];
    entities: Entity[];
  };
}